src/AssemblyGenerator.Types.cs

```diff
@@ -155,6 +156,23 @@ private void CreateType(Type type, List<DelayedWrite> genericParams)
             CreateMethods(type.GetMethods(AllMethods), genericParams);
         }
 
+        private static string Unescape(string str)
+        {
+            int idx = str.IndexOf('\\');
+            if (idx < 0)
+                return str;
+            StringBuilder sb = new StringBuilder();
```

Comment:
Why not just use `str.Replace("\\", String.Empty)` - AFAIU we just remove all `\` symbols from input string, right? You don't even need the first check if you'll use this approach. Because `Replace` returns the original string if no replacement is performed.